internal/error.go

```diff
@@ -460,6 +464,7 @@ func (wc *workflowEnvironmentInterceptor) NewContinueAsNewError(
 		WorkflowExecutionTimeout: options.WorkflowExecutionTimeout,
 		WorkflowRunTimeout:       options.WorkflowRunTimeout,
 		WorkflowTaskTimeout:      options.WorkflowTaskTimeout,
+		VersioningIntent:         options.VersioningIntent,
```

Comment:
Are we keeping this in the context or adding a new constructor for the CAN error?
I prefer the latter.

Reply:
I do too but that ship has sailed, we'll just keep it consistent here.

Reply:
Manual construction works for me.